app/pages/reset-password/new-password-form.spec.js

```diff
@@ -5,15 +5,41 @@ import sinon from 'sinon';
 import NewPasswordForm from './new-password-form';
 
 describe('NewPasswordForm', () => {
+  let wrapper;
+
+  before(() => {
+    wrapper = shallow(<NewPasswordForm />);
+  });
+
   it('renders a new password form for a user', () => {
-    const wrapper = shallow(<NewPasswordForm />);
     assert.equal(wrapper.find('form').length, 1);
   });
 
   it('should call onSubmit when user submits a new password', () => {
     const newPasswordSpy = sinon.spy();
-    const wrapper = mount(<NewPasswordForm onSubmit={newPasswordSpy} />);
+    wrapper = mount(<NewPasswordForm onSubmit={newPasswordSpy} />);
     wrapper.find('button').simulate('submit');
     assert.equal(newPasswordSpy.calledOnce, true);
   });
+
+  it('conditionally shows and hides an in-progress icon', () => {
+    wrapper.setProps({ inProgress: true });
+    assert.equal(wrapper.find('i').length, 1);
+    wrapper.setProps({ inProgress: false });
+    assert.equal(wrapper.find('i').length, 0);
+  });
+
+  it('conditionally shows and hides a reset success icon', () => {
+    wrapper.setProps({ resetSuccess: true });
+    assert.equal(wrapper.find('i').length, 1);
+    wrapper.setProps({ resetSuccess: false });
+    assert.equal(wrapper.find('i').length, 0);
+  });
+
+  it('conditionally shows and hides a reset password error message', () => {
+    wrapper.setProps({ resetError: 'test error message' });
+    assert.equal(wrapper.find('Translate').last().prop('content'), 'resetPassword.resetError');
+    wrapper.setProps({ resetError: null });
+    assert.notEqual(wrapper.find('Translate').last().prop('content'), 'resetPassword.resetError');
```

Comment:
This doesn't exactly tell us that the error message isn't displaying, but just that the last Translate component found doesn't equal the expected content. One way to clarify this is to maybe test the number of Translate components expected.